ibm/service/cos/resource_ibm_cos_bucket_lifecycle_configuration.go

```diff
@@ -0,0 +1,568 @@
+	d.SetId(bktID)
+	return resourceIBMCOSBucketLifecycleConfigurationUpdate(d, meta)
+}
+
+func resourceIBMCOSBucketLifecycleConfigurationUpdate(d *schema.ResourceData, meta interface{}) error {
+	bucketCRN := d.Get("bucket_crn").(string)
+	bucketName := strings.Split(bucketCRN, ":bucket:")[1]
+	instanceCRN := fmt.Sprintf("%s::", strings.Split(bucketCRN, ":bucket:")[0])
+	bucketLocation := d.Get("bucket_location").(string)
+	endpointType := d.Get("endpoint_type").(string)
+	bxSession, err := meta.(conns.ClientSession).BluemixSession()
+	if err != nil {
+		return err
+	}
+	s3Client, err := getS3ClientSession(bxSession, bucketLocation, endpointType, instanceCRN)
+	if d.HasChange("lifecycle_rule") {
+		lifecycleRule := d.Get("lifecycle_rule")
+		rules := lifecycleConfigurationSet(lifecycleRule.([]interface{}))
+		if err != nil {
+			return fmt.Errorf("Failed to read lifecycle configuration for COS bucket %s, %v", bucketName, err)
+		}
+		putBucketLifecycleConfigurationInput := s3.PutBucketLifecycleConfigurationInput{
+			Bucket: aws.String(bucketName),
+			LifecycleConfiguration: &s3.LifecycleConfiguration{
+				Rules: rules,
+			},
+		}
+		_, err = s3Client.PutBucketLifecycleConfiguration(&putBucketLifecycleConfigurationInput)
+
+		if err != nil {
+			return fmt.Errorf("Failed to put Lifecycle configuration on the COS bucket %s, %v", bucketName, err)
+		}
+	}
+	return resourceIBMCOSBucketLifecycleConfigurationRead(d, meta)
+}
+
+func resourceIBMCOSBucketLifecycleConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	bucketCRN := parseWebsiteId(d.Id(), "bucketCRN")
+	bucketName := parseWebsiteId(d.Id(), "bucketName")
+	bucketLocation := parseWebsiteId(d.Id(), "bucketLocation")
+	instanceCRN := parseWebsiteId(d.Id(), "instanceCRN")
+	endpointType := parseWebsiteId(d.Id(), "endpointType")
+	d.Set("bucket_crn", bucketCRN)
+	d.Set("bucket_location", bucketLocation)
+	if endpointType != "" {
+		d.Set("endpoint_type", endpointType)
+	}
+	bxSession, err := meta.(conns.ClientSession).BluemixSession()
+	if err != nil {
+		return err
+	}
+	s3Client, err := getS3ClientSession(bxSession, bucketLocation, endpointType, instanceCRN)
+	if err != nil {
+		return err
+	}
+	//getBucketConfiguration
+	const (
+		lifecycleConfigurationRulesSteadyTimeout = 2 * time.Minute
+	)
+	getLifecycleConfigurationInput := &s3.GetBucketLifecycleConfigurationInput{
+		Bucket: aws.String(bucketName),
+	}
+	var output *s3.GetBucketLifecycleConfigurationOutput
+
+	// Adding a retry to overcome the
+	err = resource.Retry(lifecycleConfigurationRulesSteadyTimeout, func() *resource.RetryError {
+		var err error
+		output, err = s3Client.GetBucketLifecycleConfiguration(getLifecycleConfigurationInput)
+
+		if d.IsNewResource() && err != nil && strings.Contains(err.Error(), "NoSuchLifecycleConfiguration: The lifecycle configuration does not exist") {
+
```

Comment:
remove blank line here. This applies to all the instances where ever extra blank lines are present

Reply:
done.